src/ai/models/porygon_model/porygon_model.py

```diff
@@ -39,5 +16,16 @@ class PorygonModel(ModelInterface):
     """
 
     def __init__(self):
-        self.take_turn = take_turn
-        self.force_switch_pokemon = force_switch_pokemon
+        pass
+
+    def take_turn(self, player: Player, other_player: Player, attack: Callable[[Move], None], use_item: Callable[[Item], None], switch_pokemon_at_idx: Callable[[int], None]) -> None:
```

Comment:
Mind making every model––super genius, random, sample, and model interface––also have non-static `take_turn` and `force_switch_pokemon` methods?